### PR TITLE
ENH: GeoDataFrame.from_features to accept FeatureCollection as well

### PR DESCRIPTION
**Pull request by emiliom**
Closes https://github.com/geopandas/geopandas/issues/225
See [jupyter notebook with usage examples](https://gist.github.com/emiliom/bde588461f816a0f0b5e4ae47ffbcc20)

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Thanks! Can you add some tests for this? (for the different cases)

**Comment by emiliom** [comment]
> Can you add some tests for this? (for the different cases)

Not this week, sorry. I could try next week.

**Comment by emiliom** [comment]
@jorisvandenbossche, are you holding off to merge until the tests are in place? I was hoping this PR could make it into the next release. The code change is actually pretty small and simple :smiley:

**Comment by jorisvandenbossche** [comment]
Yes, we normally don't merge PRs without tests. But based on the example from the notebook, I think it should not that much of work to add some tests.

Copying from there:
```
data = {'name': ['a', 'b', 'c'],
        'lat': [45, 46, 47.5],
        'lon': [-120, -121.2, -122.9]}

df = pd.DataFrame(data)
geometry = [Point(xy) for xy in zip(df['lon'], df['lat'])]
gdf = GeoDataFrame(df, geometry=geometry)

# test FeatureCollection
res = GeoDataFrame.from_features(gdf.__geo_interface__)
assert_fram_equal(res, gdf)

# test list of Features
res = GeoDataFrame.from_features(gdf.__geo_interface__['features'])
assert_fram_equal(res, gdf)

# test __geo_interface__ attribute (a GeoDataFrame actually has one)
res = GeoDataFrame.from_features(gdf)
assert_fram_equal(res, gdf)
```

**Comment by emiliom** [comment]
Thanks @jorisvandenbossche 

@lsetiawan or @ocefpaf, could you help out creating those tests?? I realize I'm pulling you into this w/o much background :frowning_face: (specially @ocefpaf) -- please see the links at the start of this PR. But I'll be out today and tomorrow, and you two are much better than me at working with test frameworks!

**Comment by jorisvandenbossche** [comment]
No worries, I will quickly add the tests

File: geopandas/geodataframe.py [hunk 1]
```diff
@@ -172,13 +172,43 @@ def from_file(cls, filename, **kwargs):
...
+        crs : str or dict (optional)
+            Coordinate reference system to set on the resulting frame.
+
+        Returns
+        -------
+        GeoDataFrame
+
+        Notes
+        -----
+        For more information about the ``__geo_interaface__``, see
```
Comment (jdmcbr) [inline geopandas/geodataframe.py]: small typo: ``__geo_interface__``

**Comment by jdmcbr** [comment]
Aside from the extremely minor comment above, looks good to me.

**Comment by jorisvandenbossche** [comment]
@emiliom Thanks!

**Comment by emiliom** [comment]
Thank you, @jorisvandenbossche!

**Pull request merged**